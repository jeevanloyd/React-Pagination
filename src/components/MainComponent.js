import React, { useEffect, useState } from "react";
import ReactPaginate from "react-paginate";
import { useDispatch } from "react-redux";
import PageCount from "./PageCount";
import ProductsDisplay from "./ProductsDisplay";


const MainComponent = () => {
    const [data, setdata] = useState([]);
    const [pageCount, setpageCount] = useState(0);
    const [limit, setLimit] = useState(8);
    const [total, setTotal] = useState(0);
    const dispatch = useDispatch()

    useEffect(() => {
        const getData = async () => {
            const res = await fetch(
                `http://localhost:4000/Data?_page=1&_limit=${limit}`
            );
            const data = await res.json();
            const total = res.headers.get("x-total-count");
            setTotal(total)
            setpageCount(Math.ceil(total / limit));
            setdata(data);
            dispatch({ type: "add_product", product: data })
        };

        getData();
    }, [dispatch, limit]);

    const fetchData = async (currentPage) => {
        const res = await fetch(
            `http://localhost:4000/Data?_page=${currentPage}&_limit=${limit}`
        );
        const data = await res.json();
        return data;
    };

    const handleClick = async (data) => {
        let currentPage = data.selected + 1;
        const dataRecieved = await fetchData(currentPage);
        setdata(dataRecieved)
        dispatch({ type: "add_product", product: data })
    };
    return (
        <>
            <div className="row d-flex justify-content-between mt-3">
                <div style={{ width: "auto" }}>{total} Products</div>
                < PageCount limit={limit} setLimit={setLimit} />
            </div>
            <hr />
            <div className="row"  >
                {data.map((item) => {
                    return (
                        <ProductsDisplay data={item} />
                    );
                })}
            </div>

            <hr />
            <ReactPaginate
                previousLabel={"previous"}
                nextLabel={"next"}
                breakLabel={"..."}
                pageCount={pageCount}
                marginPagesDisplayed={2}
                pageRangeDisplayed={2}
                onPageChange={handleClick}
                containerClassName={"pagination justify-content-end"}
                nextClassName={"page-item"}
                nextLinkClassName={"page-link"}
                pageLinkClassName={"page-link"}
                previousClassName={"page-item"}
                pageClassName={"page-item"}
                previousLinkClassName={"page-link"}
                breakClassName={"page-item"}
                breakLinkClassName={"page-link"}
                activeClassName={"active"}
            />
        </>
    );
};
export default MainComponent;