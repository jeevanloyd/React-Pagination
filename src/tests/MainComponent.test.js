import MainComponent from '../components/MainComponent';
import renderer from 'react-test-renderer';
jest.mock('../components/MainComponent', () => () => 'MainComponent');

describe('Render Main Component', () => {
    let productData = {
        Data: [
            {
                id: 1,
                price: "$87.68",
                product_name: "Amitriptyline Hydrochloride",
                description: "synergize efficient metrics",
                product_image: "http://dummyimage.com/307x328.bmp/ff4444/ffffff"
            },
            {
                id: 2,
                price: "$58.53",
                product_name: "Zicam",
                description: "repurpose world-class metrics",
                product_image: "http://dummyimage.com/345x342.jpg/dddddd/000000"
            },
        ]
    }
    it('snapshot testing for main component', () => {
        const component = renderer.create(
            <MainComponent props={productData} />,
        );
        let ProductDisplayTree = component.toJSON();
        expect(ProductDisplayTree).toMatchSnapshot();
    })

})

