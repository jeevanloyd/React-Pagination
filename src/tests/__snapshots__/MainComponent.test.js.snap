// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Main Component snapshot testing for main component 1`] = `"MainComponent"`;
